### PR TITLE
Added a context-aware fuzzer for xss-filters

### PR DESCRIPTION
**Pull request by c0nrad**
Checkout xss-filters/test/fuzz/README.md for details.

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

File: tests/fuzz/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+```
+redis:
+	testcases:1 -> "></div><img onerror=alert(1) src=x>"
+	testcases:2 -> "alert(1)"
+	testcases:3 -> "`'`;';`alert(1)"
+
+	testcases_counter: 4
+```
+
+The fuzzer then spawns a headless browser (zombie.js), which requests the newly created test case. An example page would be localhost:1339/testcase/3
```
Comment (neraliu) [inline tests/fuzz/README.md]: why do u pick zombie.js? it is not a real headless browser if u compare with phantomjs as counterpart.

**Comment by yahoocla** [comment]
CLA is valid!

File: tests/fuzz/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# fuzzer
+
+A quick little fuzzer for detecting issues within each of the five context areas.
+
+# How it works
+
+The fuzzer creates testcases by mutating entries within [bin/seed.xss](bin/seed.xss). The entries are then stored into redis as:
```
Comment (neraliu) [inline tests/fuzz/README.md]: would like to know where does the list collect from? it looks comprehensive

File: tests/fuzz/app.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  "<input value=\"" + xssFilters.inDoubleQuotedAttr(fuzz) + "\"/>" +
+  "</div>" +
+
+
+  "<div style='border: 1px solid'> " +
+  "<h2> xssFilters.inUnQuotedAttr(fuzz) </h2> " +
+  "<input value=" + xssFilters.inUnQuotedAttr(fuzz) + "'/>" +
+  "</body> </html>";
+}
+
```
Comment (adon-at-work) [inline tests/fuzz/app.js]: I see you're testing 5 filters only. Are you aware of those URI ones?

File: tests/fuzz/README.md (hunk 1 above)
Comment (c0nrad) [inline tests/fuzz/README.md]: It was just the simplest to use.

File: tests/fuzz/README.md (hunk 2 above)
Comment (c0nrad) [inline tests/fuzz/README.md]: I just scraped from a bunch of different lists (via Googling) and modified them a tiny bit.

File: tests/fuzz/app.js (hunk 3 above)
Comment (c0nrad) [inline tests/fuzz/app.js]: Yeah, at first I figured I'd test the ones that are simplest to trigger with an XSS with. I'll look into adding the others.

**Comment by adon-at-work** [comment]
looks good to get it merged

**Comment by adon-at-work** [comment]
Hi @c0nrad, please further enhance it by following the open-source practices, like adding LICENSE file and source code headers.

**Pull request merged**